Here's the failure on macOS:

```
******************* TEST 'Swift(macosx-x86_64) :: Driver/linker-library-with-space.swift' FAILED ********************
Script:
--
: 'RUN: at line 1';   env SDKROOT=/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk '/Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/buildbot_incremental/swift-macosx-x86_64/bin/swiftc' -toolchain-stdlib-rpath -Xlinker -rpath -Xlinker /usr/lib/swift -module-cache-path /Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/buildbot_incremental/swift-macosx-x86_64/swift-test-results/x86_64-apple-macosx10.9/clang-module-cache -swift-version 4  -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.5:macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0' -sdk "" -driver-print-jobs -target x86_64-unknown-linux-gnu -Ffoo -Fsystem car -F cdr -framework bar -Lbaz -lboo -Xlinker -undefined /Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/swift/test/Driver/linker-library-with-space.swift 2>&1 > /Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/Driver/Output/linker-library-with-space.swift.tmp.linux.txt
: 'RUN: at line 2';   /Applications/Xcode-beta.app/Contents/Developer/usr/bin/python3 /Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/swift/utils/PathSanitizingFileCheck --sanitize BUILD_DIR=/Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/buildbot_incremental/swift-macosx-x86_64 --sanitize SOURCE_DIR=/Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/swift --use-filecheck /Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/buildbot_incremental/llvm-macosx-x86_64/bin/FileCheck  -check-prefix LINUX-lib-flag-space /Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/swift/test/Driver/linker-library-with-space.swift < /Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/Driver/Output/linker-library-with-space.swift.tmp.linux.txt
--
Exit Code: 1

Command Output (stderr):
--
/Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/swift/test/Driver/linker-library-with-space.swift:18:26: error: LINUX-lib-flag-space: expected string not found in input
// LINUX-lib-flag-space: -o linker
                         ^
<stdin>:3:590: note: scanning from here
/Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/buildbot_incremental/llvm-macosx-x86_64/./bin/clang -fuse-ld=gold -pie -Xlinker -rpath -Xlinker usr/lib/swift/linux usr/lib/swift/linux/x86_64/swiftrt.o /var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/linker-library-with-space-146e20.o -F foo -iframework car -F cdr @/var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/linker-library-with-space-92c32a.autolink -L usr/lib/swift/linux -lswiftCore --target=x86_64-unknown-linux-gnu -framework bar -L baz -lboo -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -undefined -o main
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             ^

Input file: <stdin>
Check file: /Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/swift/test/Driver/linker-library-with-space.swift

-dump-input=help explains the following input dump.

Input was:
<<<<<<
          1: BUILD_DIR/bin/swift-frontend -frontend -c -primary-file SOURCE_DIR/test/Driver/linker-library-with-space.swift -target x86_64-unknown-linux-gnu -disable-objc-interop -F foo -Fsystem car -F cdr -module-cache-path BUILD_DIR/swift-test-results/x86_64-apple-macosx10.9/clang-module-cache -swift-version 4 -define-availability "SwiftStdlib 5.5:macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0" -module-name main -o /var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/linker-library-with-space-146e20.o
          2: BUILD_DIR/bin/swift-autolink-extract /var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/linker-library-with-space-146e20.o -o /var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/linker-library-with-space-92c32a.autolink
          3: /Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/buildbot_incremental/llvm-macosx-x86_64/./bin/clang -fuse-ld=gold -pie -Xlinker -rpath -Xlinker usr/lib/swift/linux usr/lib/swift/linux/x86_64/swiftrt.o /var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/linker-library-with-space-146e20.o -F foo -iframework car -F cdr @/var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/linker-library-with-space-92c32a.autolink -L usr/lib/swift/linux -lswiftCore --target=x86_64-unknown-linux-gnu -framework bar -L baz -lboo -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -undefined -o main
check:18                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  X~~~~~~ error: no match found
>>>>>>
```

--- a/test/Driver/linker-library-with-space.swift
+++ b/test/Driver/linker-library-with-space.swift
@@ -15,4 +15,4 @@
 // LINUX-lib-flag-space-DAG: -L baz
 // LINUX-lib-flag-space-DAG: -lboo
 // LINUX-lib-flag-space-DAG: -Xlinker -undefined
-// LINUX-lib-flag-space: -o linker
+// LINUX-lib-flag-space: -o main

